### PR TITLE
feat(NODE-4241): add support for rewrapManyDataKey and keyMaterial

### PR DESCRIPTION
**Pull request by addaleax**
spec PR: https://github.com/mongodb/specifications/pull/1213

### CODE REVIEW

File: bindings/node/index.d.ts
Comment (durran) [inline bindings/node/index.d.ts:306]: The spec doesn't have this as optional - is there a case where it's not provided?

**Review by durran — CHANGES_REQUESTED** [review]
Spec test changes part of NODE-4110.

@addaleax Do we need to wait on the spec to merge first before merging this?

**Comment by addaleax** [comment]
> @addaleax Do we need to wait on the spec to merge first before merging this?

@durran Up to you – it would make our lives a bit easier if it's released earlier, I would be fine with it as long as everything is tagged as `@experimental` anyway 🙂

**Review by durran — APPROVED** [review]

**Pull request merged**
